### PR TITLE
Jason dev branch

### PR DESCRIPTION
**Pull request by MadeInItaly99**
Review for the 2d conversion

### CODE REVIEW

**Review by coma6440 — COMMENTED** [review]
Line 5 of main_2dconversion.m has addpath('C:\Users\Jason\Documents\ASEN4018\git\cast\main'), which is not applicable to everyone

File: main/CWcomparison.m [hunk 1]
```diff
@@ -36,17 +36,26 @@
...
 curviCWsolution = CWsolution;
 CW2solution = zeros(3,length(ref_t));
+CWsolution2 = zeros(size(hillVecs));
 for j = 1:length(ref_t)
  t = ref_t(1) - ref_t(j);
  hill_matrix = make_hill_matrix(n, t);
  CWsolution(:, j) = hill_matrix * hillVecs(:, j);
  curviCWsolution(:, j) = hill_matrix * curvHillVecs(:, j);
  CW2solution(:,j) = cw_2nd_order(hillVecs(:,j),n,rad+1200,t);
+ if j>1 && j<length(ref_t)-1
```
Comment (aacorn) [inline main/CWcomparison.m:46]: This loop only propagates a single solution (the worst one I believe) and doesn't use the small step hill matrix.

File: 2d_motion/convert_2d.m [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+% tRef: Time vector for the orbits
+% refOrbit: Cartesian state vectors of the reference orbit
+% conjOrbit: Cartesian state vectors of conjunction orbit
+% Outputs:
+% relScalledHill: relitive state vector in 2D testbead hill frame
+% tRef: Time vector
+% Author: Jason Balke | Project: CAST | Date: 10/20/20
+%----------------------------------------------------------------------------------------%
+% Get normal vector to collision plane and location of collision
+planeUnitVec = refOrbit(1,1:3)/norm(refOrbit(1,1:3));
```
Comment (aacorn) [inline 2d_motion/convert_2d.m:15]: Collision is defined by the relative velocity unit vector.

File: 2d_motion/main_2dconversion.m [hunk 3]
```diff
@@ -2,41 +2,42 @@
 % This is a script that makes the 2d motion, it'll become a function at
 % some point
 clear;close all;clc;
-%% Generate Orbits
+addpath('C:\Users\Jason\Documents\ASEN4018\git\cast\main')
```
Comment (aacorn) [inline 2d_motion/main_2dconversion.m:5]: addpath('../main') would be appropriate, better would be to move the main script into the main folder and just call the importCast script.

File: analysis/planar_dev/ThreeD2TwoD.m [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+Omega_mag = Vmag/R_orbit; %rad/s
+T = 2*pi*sqrt(R_orbit^3/mu);   %orbital period - s
+tprop = 0:.5:T;
+
+Ri = R_orbit * [0;0;1];  %zdir = R_orbit
+thetas = deg2rad(linspace(-135,-45,10)); %collision angles
+subplot(1,2,1)
+
+%plot 500 second interval for each collision angle
+for i = 1:10
```
Comment (aacorn) [inline analysis/planar_dev/ThreeD2TwoD.m:30]: where do the bounds on this loop come from?

File: analysis/planar_dev/ThreeD2TwoD.m [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    Vi_obj1 = [1;0;0]*Vmag;%since z is at it's max value, must be restricted to x-y plane
+    Vi_obj2 = [x;y;0]*Vmag;
+    
+    State1 = [Ri;Vi_obj1];
+    State2 = [Ri;Vi_obj2];
+    [error100(i),angles100(i,:)] = plotintersect(State1, State2,100);
+    
+end
+
+function [error,dev_fromPlane] = plotintersect(State1, State2,timeplot)
```
Comment (aacorn) [inline analysis/planar_dev/ThreeD2TwoD.m:66]: functions should have their own file before being pulled in.

File: 2d_motion/convert_2d.m [hunk 6]
```diff
@@ -12,7 +12,7 @@
 % Author: Jason Balke | Project: CAST | Date: 10/20/20
 %----------------------------------------------------------------------------------------%
 % Get normal vector to collision plane and location of collision
-planeUnitVec = refOrbit(1,1:3)/norm(refOrbit(1,1:3));
+planeNormalVec = cross(refOrbit(1,4:6),conjOrbit(1,4:6));
```
Comment (aacorn) [inline 2d_motion/convert_2d.m:15]: normal vec should be (refOrbit(2, 4:6) - conjOrbit(2, 4:6)) as the relative velocity vector

**Pull request merged**